File: 2021/day01.py
with open('input_files/day01') as f:
    data = [int(x) for x in f.read().splitlines()]

print(sum(x < y for x, y in zip(data, data[1:])))

windowed = [sum(data[i:i + 3]) for i in range(len(data) - 2)]
print(sum(x < y for x, y in zip(windowed, windowed[1:])))

File: kotlin/src/main/kotlin/y2022/day01.kt
package y2022

import read

fun main() {
    val data = read("y2022/day01")
        .split("\n\n")
        .map { it.lines().sumOf { item -> item.toInt() } }
        .sortedDescending()

    println(data[0])
    println(data.take(3).sum())
}
